nyxtools/pilatus.py

```diff
@@ -0,0 +1,44 @@
+import datetime
+from ophyd.areadetector.detectors import PilatusDetector
+from ophyd import Component as Cpt
+from ophyd import Device, EpicsPathSignal, EpicsSignal, ImagePlugin, Signal
+from ophyd.areadetector.base import ADComponent
+from ophyd.areadetector.filestore_mixins import FileStoreBase
+
+class PilatusSimulatedFilePlugin(Device, FileStoreBase):
+    file_path = ADComponent(EpicsPathSignal, "FilePath", string=True, path_semantics="posix")
+    external_name = Cpt(Signal, value="")
+
+    def __init__(self, *args, **kwargs):
+        self.filestore_spec = "AD_PILATUS_MX"
+        super().__init__(*args, **kwargs)
+
+    def stage(self):
+        write_path = datetime.datetime.now().strftime(self.write_path_template)
```

Comment:
This is a local variable that is dropped when we lease the frame, did you mean to stash on the object and have it be used later?

Reply:
Yes, I am still trying to understand how the staging worked for the Eiger plugin so I can adapt it for this one. There are still pieces missing that I am filling in.